src/release-specification.ts

```diff
@@ -301,6 +300,72 @@ export async function validateReleaseSpecification(
           });
         }
       }
+
+      // Check to compel users to release packages with breaking changes alongside their dependents
+      if (
+        versionSpecifierOrDirective === 'major' ||
+        (isValidSemver(versionSpecifierOrDirective) &&
+          diff(
+            pkg.validatedManifest.version,
+            versionSpecifierOrDirective as string,
+          ) === 'major')
+      ) {
+        const missingDependents = Object.values(
+          project.workspacePackages,
+        ).filter((dependent) => {
+          const { dependencies, peerDependencies } =
+            dependent.unvalidatedManifest;
+          const isDependent =
+            (dependencies && hasProperty(dependencies, packageName)) ||
+            (peerDependencies && hasProperty(peerDependencies, packageName));
+
+          if (!isDependent) {
+            return false;
+          }
+
+          const dependentVersionSpecifierOrDirective =
+            unvalidatedReleaseSpecification.packages[
+              dependent.validatedManifest.name
+            ];
+
+          return (
+            dependentVersionSpecifierOrDirective !== SKIP_PACKAGE_DIRECTIVE &&
```

Comment:
In this case I think we do want to show this error if the user has `null` selected, because of the hazards.